src/runtime_src/core/pcie/tools/xbmgmt/firmware_image.cpp

```diff
@@ -559,6 +560,22 @@ firmwareImage::firmwareImage(const char *file, imageType type) :
                 mBuf = new char[bufsize];
                 in.seekg(flashSection->m_sectionOffset + flashMeta.m_image_offset);
                 in.read(mBuf, bufsize);
+            } 
+            else if (pdiSection) {
+                // Load entire PDI section.
+                std::shared_ptr<char> pdibuf(new char[pdiSection->m_sectionSize]);
```

Comment:
Either use std::vector<char> or std::unique_ptr<char[]>.   This is not shared ownership, plus there would be a mismatched free since new []  requires delete [].  Line 519 has same mistake.